apps/vaccines-updates-client/src/components/main-form/parts/facebook-post/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button } from '@material-ui/core';
+import { CITIES, HEALTH_CARE_SERVICES } from '@vacgaps/constants';
+import React, { FunctionComponent, useCallback, useState } from 'react';
+import styled from 'styled-components';
+import { useGetReports } from '../../../../hooks/useGetReports';
+
+const FacebookPostContainer = styled.div`   
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    padding: 20px;
+    border: 1px solid lightgray;
+    border-radius: 20px;
+    background-color:lightgray;
+    width: 90%;
+    max-width: 450px;
+`;
+
+const FacebookPostTextArea = styled.textarea`
+    width: 90%;
+`;
+
+function addFieldToRow(rowState: { isAddedField: boolean }, fieldPrefix: string, value: string, convertValue?: (val: string) => string): string {
+    if (!value) {
+        return '';
+    }
+
+    let result = '';
+    if (!rowState.isAddedField) {
+        rowState.isAddedField = true;
+        result += ': ';
+    } else {
+        result += ', ';
+    }
+
+    result += fieldPrefix + (convertValue ? convertValue(value) : value);
+    return result;
+}
+
+const Comp: FunctionComponent<{ className?: string; }> = props => {
+    const [facebookPost, setFacebookPost] = useState<string>();
+    const [textAreaRows, setTextAreaRows] = useState<number>();
+
+    const getReports = useGetReports(/*returnHiddenReports=*/false);
+    const generatePost = useCallback(async () => {
+        // TODO: Error treatment
+        const reportsResponse = await getReports();
+        const reports = reportsResponse.reports.sort(function (a, b) {
+            let result = a.healthCareService.localeCompare(b.healthCareService);
+            // if (result === 0) result = a.serviceEndTime - b.serviceEndTime;
```

Comment:
What do you need this comment for?

Reply:
Actually not sure. We didn't consider the sorting criterions, just wanted something to be consistent enough so users will not get confused. I removed the endTIme sorting when it became optional variable.
I'll leave this comment for now just to have a commit which exactly matches the deployed version, we can clean it or add it back in the future.